pkg/service/controller.go

```diff
@@ -271,7 +271,16 @@ func (c *ControllerService) ControllerPublishVolume(
 
 	err = c.virtClient.AddVolumeToVM(c.infraClusterNamespace, vmName, addVolumeOptions)
 	if err != nil {
-		klog.Errorf("Failed adding volume %s to VM %s, %v", dvName, vmName, err)
+		// Need to ignore already exists error due to multiple add requests
+		if !strings.Contains(err.Error(), fmt.Sprintf("Unable to add volume [%s] because it already exists", dvName)) {
+			klog.Errorf("failed adding volume %s to VM %s, %v", dvName, vmName, err)
+			return nil, err
+		}
+	}
+
+	err = c.virtClient.EnsureVolumeAvailable(c.infraClusterNamespace, vmName, dvName)
```

Comment:
would it make sense to pass a time limit duration in here so it's obvious how long we're waiting?

Reply:
It really doesn't matter how long we wait, the point of this is that we shouldn't return success until the hotplugged volume is completely ready otherwise we get the can't find by serial error. I can easily add a parameter with a timeout, it doesn't matter that much